src/Grpc.Net.ClientFactory/GrpcClientServiceExtensions.cs

```diff
@@ -334,13 +334,23 @@ public static IHttpClientBuilder AddGrpcClient<TClient>(this IServiceCollection
 
             services.PostConfigure<HttpClientFactoryOptions>(name, options =>
             {
-                options.HttpMessageHandlerBuilderActions.Add(static builder =>
+                options.HttpMessageHandlerBuilderActions.Add(builder =>
                 {
                     if (builder.PrimaryHandler == null)
                     {
                         // This will throw in .NET Standard 2.0 with a prompt that a user must set a handler.
                         // Because it throws it should only be called in PostConfigure if no handler has been set.
                         var handler = HttpHandlerFactory.CreatePrimaryHandler();
+#if NET5_0_OR_GREATER
+                        if (handler is SocketsHttpHandler socketsHttpHandler)
+                        {
+                            // A channel is created once per client, lives forever, and the primary handler never changes.
+                            // It's possible that long lived connections cause the client to miss out on DNS changes.
+                            // Replicate the core benefit of a handler lifetime (periodic connection recreation)
+                            // by setting PooledConnectionLifetime to handler lifetime.
```

Comment:
Thanks @JamesNK to address this issue based on our offline discussion. Leave few comments here:
1. Since we cache channel per typed/named client (no matter the client is transient or singleton), and use PooledConnectionLifetime to control handler lifetime, we are only using httpclientfactory as configuration API, the httpclientfactory handler pool is redundant in this scenario. I think we should work with[ runtime team to make httpclientfactory just as configuration API](https://github.com/dotnet/runtime/issues/74508).
2. With caching channel and use PooledConnectionLifetime, would you mind to share more details on why we DI GrpcClientFactory as transient rather than singleton (same as DefatultHttpClientFactory)? 
` services.TryAddTransient<GrpcClientFactory, DefaultGrpcClientFactory>();`

Reply:
> With caching channel and use PooledConnectionLifetime, would you mind to share more details on why we DI GrpcClientFactory as transient rather than singleton (same as DefatultHttpClientFactory)?
services.TryAddTransient<GrpcClientFactory, DefaultGrpcClientFactory>();

So the current `IServiceProvider` scope is injected into the factory.